pandas/core/arrays/datetimelike.py

```diff
@@ -1437,6 +1437,54 @@ def max(self, axis=None, skipna=True, *args, **kwargs):
         # Don't have to worry about NA `result`, since no NA went in.
         return self._box_func(result)
 
+    def mean(self, axis=None, skipna=True):
+        """
+        Return the mean value of the Array or mean along an axis.
+
+        .. versionadded:: 0.25.0
+
+        Parameters
+        ----------
+        axis : None
+            Dummy parameter to match NumPy signature
+        skipna : bool, default True
+            Whether to ignore any NaT elements
+
+        See Also
+        --------
+        numpy.ndarray.mean
+        Series.mean : Return the mean value in a Series.
+
+        Notes
+        -----
+        mean is only defined for Datetime and Timedelta dtypes, not for Period.
+        """
+        if is_period_dtype(self):
+            # See discussion in GH#24757
+            raise TypeError(
+                "mean is not implemented for {cls} since the meaning may be "
```

Comment:
```suggestion
                "mean is not implemented for {cls} since the meaning is "
```

Reply:
The thing here is that it isn't ambiguous for Tick-frequencies, so "may be" is more accurate.

Reply:
What is the unambiguous result of the mean of this?

```
In [12]: idx = pd.period_range('2012-01-01', periods=4, freq='1s')

In [13]: idx
Out[13]: 
PeriodIndex(['2012-01-01 00:00:00', '2012-01-01 00:00:01',
             '2012-01-01 00:00:02', '2012-01-01 00:00:03'],
            dtype='period[S]', freq='S')
```

Reply:
`Period('2012-01-01 00:00:01', freq='S')`.  We already discussed rounding error in this thread.

Reply:
We mainly discussed that for datetime64, where I am fine with that. 
But for Period, you implicitly used the start_times for taking the mean, while you could also have taken the end_times. So I still find this ambiguous. 

(anyway, I think we agree to disallow it all-together for all Periods, so this is only about the error message)